cmd/k8s-operator/deploy/chart/values.yaml

```diff
@@ -111,6 +111,13 @@ proxyConfig:
 # Kubernetes API server.
 # https://tailscale.com/kb/1437/kubernetes-operator-api-server-proxy
 apiServerProxyConfig:
+  # Set to "true" to enable the ClusterRole permissions required for the API
+  # server proxy to impersonate groups and users based on tailnet ACL grants.
+  # Required to deploy ProxyGroups of type "kube-apiserver" in auth mode.
+  authEnabled: "false" # "true", "false"
+
+  # If true or noauth, the operator will run an in-process API server proxy.
+  # Deprecated: use apiServerProxyConfig.authEnabled instead.
```

Comment:
Should we maybe not add the deprected note yet? Or alternatively, we'd want to link some docs for how to use the new mechanism + add some note about when/if this field would  get removed. I'm just worried about confusing users if we leave this as is for release - the `mode` field was responsible for setting up the proxy while the `authEnabled` field seems to only create the RBAC so it's not a full replacement?

Reply:
Yeah agreed I don't think there's any reason to rush to deprecate this. The two can happily co-exist, so I'll reword these comments.